### PR TITLE
Documentation for migrating workloads to the new namespace

### PR DESCRIPTION
**Pull request by AlexandraRoatis**
## Proposed Changes

* Documentation for migrating workloads to the new namespace. In the current state we assume the use is based on `knative-gcp` and Kubernetes secret-based authentication.

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

**Comment by eclipselu** [comment]
a general comment on the structure, should we put everything under a directory `docs/how-to/namespace-transition`?

File: docs/how-to/namespace-transition.md [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+   job.batch/namespace-transition                    1/1           37s        3m3s
+   ```
+
+   ```shell script
+   # delete the job
+   kubectl delete -f namespace-transition.yaml
+   ```
+
+1. Upgrade to the new namespace by applying the changes from the `master` branch
+   or any of the releases from `v0.20` onward. This instructions are identical
```
Comment (nlopezgi) [inline docs/how-to/namespace-transition.md:80]: knative supports upgrading by a single minor version number, so these instructions, imo, should be very specific to indicate migration from 0.19 to 0.20. They should also probably not recommend using the master branch. https://knative.dev/docs/install/upgrade-installation/

File: docs/how-to/namespace-transition.md [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+   ```shell script
+   # apply changes based on the master branch
+   cd knative-gcp
+   git checkout master
+   ko apply -f ./config
+   ```
+
+   ```shell script
+   # apply changes based on the release
+   export KGCP_VERSION=v0.20.0 # or above
```
Comment (nlopezgi) [inline docs/how-to/namespace-transition.md:93]: Note below these will work only once 0.20 is released, remove the # or above (As per my comment above)

File: docs/how-to/namespace-transition.md [hunk 3]
```diff
@@ -0,0 +1,286 @@
+# Transition a project to the new `events-system` namespace
```
Comment (grantr) [inline docs/how-to/namespace-transition.md:1]: Would cluster make more sense than project here?

File: docs/how-to/namespace-transition.md [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+release from the `knative-gcp` repository. We will update the page with specific
+`gcloud` instructions as the release becomes available there as well.**
+
+## Recommended Migration
+
+1. Run the following commands before installing the new `v0.20` release. We
+   explain the purpose of each command in comments.
+
+   ```shell script
+   # create the new namespace by copying the information from the old one
```
Comment (grantr) [inline docs/how-to/namespace-transition.md:23]: Can we be more specific than "information"? Is the purpose of this line to copy labels and annotations?

Comment (AlexandraRoatis) [reply docs/how-to/namespace-transition.md]: Yes. I updated the comment.

File: docs/how-to/namespace-transition.md (hunk 3 above)
Comment (AlexandraRoatis) [reply docs/how-to/namespace-transition.md]: Updated.

File: docs/how-to/namespace-transition.md (hunk 2 above)
Comment (AlexandraRoatis) [reply docs/how-to/namespace-transition.md]: Done.

File: docs/how-to/namespace-transition.md (hunk 1 above)
Comment (AlexandraRoatis) [reply docs/how-to/namespace-transition.md]: Done.

**Comment by AlexandraRoatis** [comment]
@eclipselu Since the number of files in the `how-to` directory is low, my preference would be for not changing the file structure. But, if there are any strong opinions on this matter I can make the update.

**Pull request merged**